signer/tests/integration/complete_deposit.rs

```diff
@@ -0,0 +1,742 @@
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+    let signatures_required = 3;
+
+    let signer_set = deposit_setup(&mut rng, &pg_store).await;
+    // Get the chain tip.
+    let chain_tip = get_bitcoin_canonical_chain_tip_block(&pg_store)
+        .await
+        .unwrap()
+        .unwrap();
+    // Normal: Get an existing deposit request on the canonical bitcoin
+    // blockchain.
+    let deposit_req = pg_store
+        .get_pending_accepted_deposit_requests(&chain_tip.block_hash, 20, signatures_required)
+        .await
+        .unwrap()
+        .last()
+        .cloned()
+        .unwrap();
+
+    // Normal: we generate a transaction that sweeps in the deposit.
+    let sweep_config = SweepTxConfig {
+        signer_public_key: PublicKey::combine_keys(&signer_set).unwrap(),
+        amounts: 3000..1_000_000_000,
+        inputs: vec![deposit_req.outpoint()],
+    };
+    let mut sweep_tx: model::Transaction = sweep_config.fake_with_rng(&mut rng);
+    // Normal: make sure the sweep transaction is on the canonical bitcoin
+    // blockchain and is in our database.
+    sweep_tx.block_hash = chain_tip.block_hash.into_bytes();
+    // Normal: make sure that we have a record of the sweep transaction in
+    // our database.
+    let bitcoin_tx_ref = BitcoinTxRef {
+        txid: sweep_tx.txid.into(),
+        block_hash: sweep_tx.block_hash.into(),
+    };
+    pg_store.write_transaction(&sweep_tx).await.unwrap();
+    pg_store
+        .write_bitcoin_transaction(&bitcoin_tx_ref)
+        .await
+        .unwrap();
+
+    // Generate the transaction and corresponding request context.
+    let (complete_deposit_tx, ctx) = make_complete_deposit(&deposit_req, &sweep_tx, &chain_tip);
+
+    // This should not return an Err.
+    complete_deposit_tx.validate(&pg_store, &ctx).await.unwrap();
+
+    testing::storage::drop_db(pg_store).await;
+}
+
+/// For this test we check that the `CompleteDepositV1::validate` function
+/// returns a deposit validation error with a DeployerMismatch message when
+/// the deployer doesn't match but everything else is okay.
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn complete_deposit_validation_deployer_mismatch() {
+    // Normal: this generates the blockchain as well as deposit request
+    // transactions in each bitcoin block.
+    let db_num = DATABASE_NUM.fetch_add(1, Ordering::SeqCst);
+    let pg_store = testing::storage::new_test_database(db_num, true).await;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+    let signatures_required = 3;
+
+    let signer_set = deposit_setup(&mut rng, &pg_store).await;
+
+    // Get the chain tip
+    let chain_tip = get_bitcoin_canonical_chain_tip_block(&pg_store)
+        .await
+        .unwrap()
+        .unwrap();
+    // Normal: Get an existing deposit request on the canonical bitcoin
+    // blockchain.
+    let deposit_req = pg_store
```

Comment:
You're doing this 7-liner in every test but one, maybe make a helper function to keep from detracting from the test logic in these cases?

Reply:
Good point, don't know why I noticed the other one and not this one.